sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpInvocationTracker.cs

```diff
@@ -61,6 +61,10 @@ public override bool FirstParameterIsStringAndIsNotConstant(InvocationContext co
             return !(argumentsSyntax?.Expression.IsConstant(context.Model) ?? false);
         }
 
+        public override InvocationCondition WhenMethodNameIs(string methodName) =>
+            (context) => context.Identifier is SimpleNameSyntax nameSyntax
+                        && nameSyntax.Identifier.ValueText == methodName;
+   
```

Comment:
Just checking the method name and not the type here (the hash rule is looking for any method called _Create_ that returns a specific type).